pkg/auth/nodeidentifier/default.go

```diff
@@ -55,10 +55,11 @@ func (defaultNodeIdentifier) NodeIdentity(u user.Info) (string, bool) {
 	}
 
 	userName := u.GetName()
-	nodeName := ""
-	if strings.HasPrefix(userName, nodeUserNamePrefix) {
-		nodeName = strings.TrimPrefix(userName, nodeUserNamePrefix)
+	if !strings.HasPrefix(userName, nodeUserNamePrefix) {
```

Comment:
update godoc to match and move the username check before the loop over the groups